src/com/google/enterprise/adaptor/database/DatabaseAdaptor.java

```diff
@@ -345,6 +365,9 @@ public void init(AdaptorContext context) throws Exception {
 
     uniqueKey = ukBuilder.build();
     log.config("primary key: " + uniqueKey);
+
+    docIdLastModifiedMap =
+        CacheBuilder.newBuilder().maximumSize(1000000).build();
```

Comment:
maximumSize takes a long, so add L to the value (FYI, initialCapacity takes an int).